src/lib/journey/_utilities/metadata.utilities.ts

```diff
@@ -77,12 +105,28 @@ export function isSelfSubmitting(callback: FRCallback) {
   return selfSubmittingCallbacks.includes(callback.getType());
 }
 
+/**
+ * @function isStepProvidingLocalAuth -
+ * @param {array} callbacks - CallbackMetadata
+ * @returns
+ */
+export function isStepProvidingLocalAuth(callbacks: CallbackMetadata[]) {
+  const localAuthenticationCallbacks = callbacks.filter(
+    (callback) => callback.social && callback.social.isLocalAuthenticationEnabled,
+  );
+  return localAuthenticationCallbacks.length > 0;
+}
+
 /**
  * @function isStepSelfSubmittable -
  * @param {array} callbacks - CallbackMetadata
  * @returns
  */
 export function isStepSelfSubmittable(callbacks: CallbackMetadata[]) {
+  if (callbacks.filter((callback) => callback.social)) {
+    return true;
+  }
+
```

Comment:
I found that the desired behaviour for the SelectIDPCallback is slightly different to other callbacks in terms of self-submission. An example of this is the fact that clicking one of the social provider buttons should submit the form regardless of the state of any of the other callbacks on the page - unlike other "Self Submitting" callbacks which generally need there to be no other "Unsubmittable" callbacks.

By the way when I say "desired" behaviour, I mean the expected user experience on such a page - I would expect to click "Continue With Apple" for example, and to skip anything else on the page (e.g. Username, Password).

Reply:
I think what you did on lines 156-159 of this file is reasonable. I can't think of a reason why this wouldn't work, and it's a similar pattern for `ConfirmationCallback`.

Reply:
I kind of changed my mind on this, but it was not this specifically, but just the overall approach to this architectural challenge. Let me know if this comment makes sense and how it affects what you wrote: https://github.com/cerebrl/forgerock-web-login-framework/pull/34/files#r1035288585.